codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/RuntimeTypes.kt

```diff
@@ -203,10 +204,10 @@ data class RuntimeType(val name: String?, val dependency: RustDependency?, val n
         fun forInlineDependency(inlineDependency: InlineDependency) =
             RuntimeType(inlineDependency.name, inlineDependency, namespace = "crate")
 
-        fun forInlineFun(name: String, module: String, func: (RustWriter) -> Unit) = RuntimeType(
+        fun forInlineFun(name: String, module: RustModule, func: (RustWriter) -> Unit) = RuntimeType(
```

Comment:
👍🏻 I wonder if we should add convenience constructors to `RustModule` like `.public` and `.private`